packages/terra-functional-testing/src/config/wdio.conf.js

```diff
@@ -157,12 +159,43 @@ exports.config = {
   // Test reporter for stdout.
   // The only one supported by default is 'dot'
   // see also: https://webdriver.io/docs/dot-reporter.html
-  reporters: ['spec', AccessibilityReporter],
+  reporters: ['spec', AccessibilityReporter, SpecReporter],
   //
   // Options to be passed to Mocha.
   // See the full list at http://mochajs.org/
   mochaOpts: {
     ui: 'bdd',
     timeout: 60000,
   },
+  /**
+   * Gets executed once before all workers get launched.
+   * @param {Object} config - wdio configuration object.
+   * @param {Array.<Object>} capabilities - List of capabilities details.
+   */
+  onPrepare(config) {
+    const { reporterOptions = {} } = config;
+    const { outputDir } = reporterOptions;
+
+    // Remove previous reporter results.
+    cleanResults(outputDir);
+  },
+  /**
+   * Gets executed after all workers have shut down and the process is about to exit.
+   * An error thrown in the `onComplete` hook will result in the test run failing.
+   * @param {Object} exitCode - 0 - success, 1 - fail.
+   * @param {Object} config - wdio configuration object.
+   * @param {Array.<Object>} capabilities - List of capabilities details.
+   * @param {<Object>} results - Object containing test results.
+   */
+  onComplete(_exitCode, config) {
+    const { reporterOptions = {} } = config;
+    const { outputDir } = reporterOptions;
+
+    // Merge reporter results.
+    mergeResults(outputDir, {
```

Comment:
Could you add an event listen for `onRunnerEnd`  to manage this would needing to add it in the wdio config's onComplete hook?

Reply:
No, because of how WDIO 6 initializes the reporters this work needs to happen in the configuration lifecycle hooks. In WDIO 6 a reporter is created per spec file. So any work done in a reporters `onRunnerEnd` lifecycle hooks runs per spec file, not per test run. So if there are 10 spec files, `onRunnerEnd` will be run 10 times with different instances on a reporter.

The [json reporter](https://webdriver.io/docs/docs/wdio-json-reporter.html#result-files) has a pretty good description of this:

> With WDIO v5, reporting has moved from a centralized process to one that is handled by each of the "sessions" spun up for parallel test execution.
This change helped reduce the amount of chatter during WDIO test execution and thus improved performance. The downside is we are no longer able to get a single report for ALL test execution. Consider the following:

> 2 suites of tests configured to run in 2 browsers:

> WDIO v4: 1 json file with execution results
> WDIO v5: 4 json files with execution results